### PR TITLE
Deprecate crate

### PR DESCRIPTION
**Pull request by ali-behjati**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Pyth Client
 
+**This crate has deprecated. Please use [pyth-sdk-solana](https://github.com/pyth-network/pyth-sdk-rs/tree/main/pyth-sdk-solana) instead.**
```
Comment (jayantk) [inline README.md:3]: I suggest mentioning that the other crate is ~identical, so it's very easy to migrate. If it's not trivial to migrate, we can add some text about what changed.

**Review by jayantk — COMMENTED** [review]
i don't know if you need to deprecate the individual methods or not. aside from that lgtm

**Review by Reisen — CHANGES_REQUESTED** [review]
It turns out you can deprecate the whole file in one swoop using a top-level `#![deprecated]`, I think rather than cherry picking some functions as is currently here the single top-level can be added instead.

File: README.md (hunk 1 above)
Comment (ali-bahjati) [reply README.md:3]: How does it look like now?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # Pyth Client
 
+**This crate has deprecated. Please use [pyth-sdk-solana](https://github.com/pyth-network/pyth-sdk-rs/tree/main/pyth-sdk-solana) instead. pyth-sdk-solana provides identical functionalities with an easier interface.**
```
Comment (Reisen) [inline README.md:3]: > This crate has **been** deprecated

File: src/lib.rs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 //! A Rust library for consuming price feeds from the [pyth.network](https://pyth.network/) oracle on the Solana network.
 //!
 //! Please see the [crates.io page](https://crates.io/crates/pyth-client/) for documentation and example usage.
+#![deprecated = "This crate has deprecated. Please use pyth-sdk-solana"]
```
Comment (Reisen) [inline src/lib.rs:4]: Ditto

**Review by Reisen — CHANGES_REQUESTED** [review]

**Review by Reisen — APPROVED** [review]

**Pull request merged**